middle_end/flambda/flambda_utils.ml

```diff
@@ -350,9 +350,14 @@ let make_closure_declaration
       ~body ~stub ~dbg:Debuginfo.none ~inline:Default_inline
       ~specialise:Default_specialise ~is_a_functor:false
       ~closure_origin:(Closure_origin.create (Closure_id.wrap id))
+      ~poll:Default_poll
   in
-  assert (Variable.Set.equal (Variable.Set.map subst free_variables)
+  begin
+    (* this is required because we hardcode poll behaviour above *)
+    assert( stub );
```

Comment:
Should this `~stub` parameter just be removed?

Reply:
This is probably better than asserting. I'll remove it.

Reply:
Done. Also rebased.